litmus-portal/backend/auth/pkg/server/server.go

```diff
@@ -38,8 +44,79 @@ func NewServer(cfg *Config) *Server {
 
 // Server Provide authorization server
 type Server struct {
-	Config  *Config
-	Manager *manage.Manager
+	Config      *Config
+	Manager     *manage.Manager
+	globalToken *oauth2.Token
+	config      oauth2.Config
+}
+
+//Middleware redirects to a github endpoint to get the temp code for oauth
+func (s *Server) Middleware(c *gin.Context) {
+	s.config = oauth2.Config{
```

Comment:
Same here, remove these api details if not dummy. Also this config is being repeated, you can store it in a struct and reuse it.

Reply:
We can move initalizing this config in the `NewServer()` function